app/gosqs/receive_message_test.go

```diff
@@ -0,0 +1,364 @@
+package gosqs
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReceiveMessageWaitTimeEnforcedV1(t *testing.T) {
+	// create a queue
+	req, err := http.NewRequest("POST", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	form := url.Values{}
+	form.Add("Action", "CreateQueue")
+	form.Add("QueueName", "waiting-queue")
+	form.Add("Attribute.1.Name", "ReceiveMessageWaitTimeSeconds")
+	form.Add("Attribute.1.Value", "2")
+	form.Add("Version", "2012-11-05")
+	req.PostForm = form
+
+	rr := httptest.NewRecorder()
+	status, _ := CreateQueueV1(req)
+
+	assert.Equal(t, http.StatusOK, status)
+
+	// receive message ensure delay
+	req, err = http.NewRequest("POST", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	form = url.Values{}
+	form.Add("Action", "ReceiveMessage")
+	form.Add("QueueUrl", "http://localhost:4100/queue/waiting-queue")
+	form.Add("Version", "2012-11-05")
+	req.PostForm = form
+
+	rr = httptest.NewRecorder()
+
+	start := time.Now()
+	status, _ = ReceiveMessageV1(req)
+	elapsed := time.Since(start)
+
+	assert.Equal(t, http.StatusOK, status)
+	if elapsed < 2*time.Second {
+		t.Fatal("handler didn't wait ReceiveMessageWaitTimeSeconds")
+	}
+
+	// send a message
+	req, err = http.NewRequest("POST", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	form = url.Values{}
+	form.Add("Action", "SendMessage")
+	form.Add("QueueUrl", "http://localhost:4100/queue/waiting-queue")
+	form.Add("MessageBody", "1")
+	form.Add("Version", "2012-11-05")
+	req.PostForm = form
+
+	rr = httptest.NewRecorder()
+	http.HandlerFunc(SendMessage).ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got \n%v want %v",
+			status, http.StatusOK)
+	}
+
+	// receive message
+	req, err = http.NewRequest("POST", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	form = url.Values{}
+	form.Add("Action", "ReceiveMessage")
+	form.Add("QueueUrl", "http://localhost:4100/queue/waiting-queue")
+	form.Add("Version", "2012-11-05")
+	req.PostForm = form
+
+	rr = httptest.NewRecorder()
+
+	start = time.Now()
+	status, _ = ReceiveMessageV1(req)
+	elapsed = time.Since(start)
+
+	assert.Equal(t, http.StatusOK, status)
```

Comment:
Now that we have successfully received the message, I'd like to assert that the message body is exactly `1`.

Reply:
Done!